mlflow/tracking/fluent.py

```diff
@@ -2131,6 +2131,23 @@ def initialize_logged_model(
     return model
 
 
+@contextlib.contextmanager
+def _logged_model_cm(model: LoggedModel) -> Generator[LoggedModel, None, None]:
+    """
+    Context manager to wrap a LoggedModel and update the model
+    status after the context is exited.
+    If any exception occurs, the model status is set to FAILED.
+    Otherwise, it is set to READY.
+    """
+    try:
+        yield model
+    except Exception:
+        finalize_logged_model(model.model_id, LoggedModelStatus.FAILED)
+        raise
+    else:
+        finalize_logged_model(model.model_id, LoggedModelStatus.READY)
```

Comment:
can we add a test to ensure this works?

Reply:
https://github.com/mlflow/mlflow/blob/e8147dbd14df24a365ab60e4aecdb7223fac8fac/tests/models/test_model.py#L686 is the test